Source/WebCore/style/ElementRuleCollector.cpp

```diff
@@ -584,14 +584,59 @@ bool ElementRuleCollector::containerQueriesMatch(const RuleData& ruleData, const
 
 bool ElementRuleCollector::scopeRulesMatch(const RuleData& ruleData, const MatchRequest& matchRequest)
 {
-    auto queries = matchRequest.ruleSet.scopeRulesFor(ruleData);
+    auto rules = matchRequest.ruleSet.scopeRulesFor(ruleData);
 
-    if (queries.isEmpty())
+    if (rules.isEmpty())
+        return true;
+
+    SelectorChecker checker(element().rootElement()->document());
+    SelectorChecker::CheckingContext context(SelectorChecker::Mode::CollectingRulesIgnoringVirtualPseudoElements);
+
+    auto evaluate = [&](auto& rule) {
+        Element* olderAncestorToMatch = nullptr;
```

Comment:
"Older" is not the best way to describe this. This seems to be the scope start element, not sure if there is a spec term.
This can be const.